src/renderers/webgl/loaders/AssetLoader.ts

```diff
@@ -0,0 +1,60 @@
+import { Publisher, Subscriber } from './PubSub'
+import { Assets, Texture } from 'pixi.js'
+import { noop } from '../../../utils/helpers'
+
+type LoadAssetProps = Partial<Subscriber<Texture>> & { url: string }
+
+export default class AssetLoader {
+  private cache: { [url: string]: Publisher<Texture> } = {}
+
+  static get(url: string): Texture | null {
+    if (Assets.cache.has(url)) {
+      return Assets.cache.get(url)
+    } else {
+      return null
+    }
+  }
+
+  load({ url, resolve = noop, reject = noop }: LoadAssetProps) {
+    if (this.cache[url] === undefined) {
+      this.cache[url] = this.createPublisher(url)
+    }
+
+    return this.cache[url].subscribe({ resolve, reject })
```

Comment:
Pretty sure I'm tracking the code path here correctly. But to confirm, if there's a cache hit, `this.cache[url].subscribe` will invoke `resolve` synchronously, yeah?